torchdata/datapipes/map/util/converter.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from typing import List, Optional
+
+from torch.utils.data import IterDataPipe, MapDataPipe
+
+
+# @functional_datapipe("to_iter_datapipe")  # This line must be kept for .pyi signature parser
```

Comment:
Wait, why it is not working? Do we need to update functional_datapipe to determine the proper DP type?

Reply:
* https://github.com/pytorch/data/issues/200

Basically, it is because the input and output type are different (from `MapDataPipe` to `IterDataPipe`). `@functional_datapipe` currently doesn't handle that well.